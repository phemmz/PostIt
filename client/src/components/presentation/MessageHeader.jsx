import React from 'react';
import PropTypes from 'prop-types';

/**
 * Home
 * @return {*} div
 */
const MessageHeader = ({ groupName }) => {
  return (
    <div className="row">
      <div className="col s12">
        <span><h5 id="welcome-header" className="green-text text-darken-4 card">
          <strong>
            { groupName }
          </strong></h5>
        </span>
      </div>
    </div>
  );
};

MessageHeader.propTypes = {
  groupName: PropTypes.string.isRequired
};

export default MessageHeader;
