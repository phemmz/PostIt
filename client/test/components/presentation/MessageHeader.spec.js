import React from "react";
import { mount } from "enzyme";
import MessageHeader from '../../../src/components/presentation/MessageHeader';
import { groups } from '../../__mockData__/dummyData';

describe("MessageHeader Component", () => {
  let props;
  let mountedMessageHeader;

  const messageHeader = () => {
    if (!mountedMessageHeader) {
      mountedMessageHeader = mount(
        <MessageHeader {...props} />
      );
    }
    return mountedMessageHeader;
  }

  beforeEach(() => {
    props = {
      groupName: groups[0].groupname
    };
    mountedMessageHeader = undefined;
  });

  it("should always render a div", () => {
    const divs = messageHeader().find("div");
    expect(messageHeader()).toMatchSnapshot();
    expect(divs.length).toBeGreaterThan(0);
  });

  it("should contain everything else that gets rendered", () => {
    const divs = messageHeader().find("div");
    const wrappingDiv = divs.first();
    expect(wrappingDiv.children()).toEqual(messageHeader().children());
  });

  it("should receive props", () => {
    expect(Object.keys(messageHeader().props()).length).toBe(1);
  });

  describe("when groupname prop is defined", () => {
    it("should display the groupname", () => {
      expect(messageHeader().find('span').text()).toBe("june fellows");
    });
  });
});

