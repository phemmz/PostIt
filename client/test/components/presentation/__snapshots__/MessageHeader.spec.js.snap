// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageHeader Component should always render a div 1`] = `
<MessageHeader
  groupName="june fellows"
>
  <div
    className="row"
  >
    <div
      className="col s12"
    >
      <span>
        <h5
          className="green-text text-darken-4 card"
          id="welcome-header"
        >
          <strong>
            june fellows
          </strong>
        </h5>
      </span>
    </div>
  </div>
</MessageHeader>
`;
